Lidar corr

nzywucka opened the pull request.

nzywucka committed:
Reading of uncalibrated reports from Magic lidar and implementation of container to keep the parameters.

nzywucka committed:
Removed print.

nzywucka committed:
In __init__.py, more parameters were added to ReportLaserContainer() and to parse_laser_info().

nzywucka committed:
__init__.py changed regarding Laser test #84 comments in ReportLaserContainer() class and parse_laser_info() function.

nzywucka committed:
Unnecessary spaces are removed from the file.

nzywucka committed:
Unnecessary spaces are removed

nzywucka committed:
Unnecessary spaces removed

nzywucka committed:
Typo removed

nzywucka committed:
If statement for self.laser = self.parse_laser_info() added

nzywucka committed:
Typo removed

nzywucka committed:
BGSamples parameter added

nzywucka committed:
The [0]s have been removed from the arrays while reading the Lidar reports. Filtering for the unique reports has been added.

nzywucka committed:
Unindent that does not match any outer indentation level has been corrected.

nzywucka committed:
Filtering for the unique reports has been corrected.

nzywucka committed:
Unused variable has been removed.

nzywucka committed:
Reading of the unnecesary parameters from the lidar reports are removed.
Filtering the unique reports is improved.

nzywucka committed:
Unused import has been removed
Reading the unique reports has been improved

nzywucka committed:
Trailing whitespace has been removed

nzywucka committed:
Reading the unique reports has been improved to keep all the unique parameters in the container.
All the parameters in the container are stored in the list of parameters now.

nzywucka committed:
The definitions of the container parameters have been improved.

nzywucka committed:
Appending the unique parameters to the container was improved.

nzywucka committed:
AttributeError has been fixed.

nzywucka committed:
Merge remote-tracking branch 'origin/master' into laser_test

nzywucka committed:
Filtering of lidar parameters has been improved.

nzywucka committed:
Filtering of lidar parameters has been improved

# Please enter the commit message for your changes. Lines starting

nzywucka committed:
test_lidar_parameters was removed

nzywucka committed:
Implementation of some tests of reading lidar parameters

nzywucka committed:
test_lidar_parameters test has been improved.

nzywucka committed:
Unnecessary comments and whitespaces have been removed.

nzywucka committed:
Implementation of some corrections of lidar parameters.

nzywucka committed:
Improvements to read external files have been added

nzywucka committed:
External files have been added

nzywucka committed:
Reading BadReport and State parameters added.

nzywucka committed:
Reading unique reports changed.

nzywucka committed:
Remove assert from tests in codacy

`# Please enter the commit message for your changes. Lines starting

nzywucka committed:
Skipping assert checks in the test files.

Trailing whitespaces and unused libraries have been removed.

nzywucka committed:
Skipping assert checks in the test files.

nzywucka committed:
Tests for reading lidar parameters have been removed.

jsitarek commented on ctapipe_io_magic/__init__.py:
```diff
@@ -92,6 +94,68 @@
 
+class ReportLaserContainer(Container):
+    """ Container for Magic laser parameters """
+    UniqueID = Field(List[Any], 'No.')
```
you should merge the latest changes from the lidar_test branch here

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+    def magic_reports_reading(self, input_file, process_run=False):
+        logger.info(f"\nInput file: {input_file}")
+        event_source = MAGICEventSource(input_file, process_run=process_run)
+        is_simulation = event_source.is_simulation
```
there are no reports in simulations

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        is_simulation = event_source.is_simulation
+        logger.info(f"\nIs simulation: {is_simulation}")
+
+        obs_id = event_source.obs_ids[0]
```
is this needed, looks like debugging

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+
+        laser = event_source.laser
+
+        for key, report_list in laser.items():
```
this does not fit the current scheme of writing the reports: list rather then a dictionary
and the whole loop is not needed because it just shown a debug message

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+                    variable, value = line.split("=")
+                    variable = variable.strip()
+                    value = value.split("#")[0].strip()
+                    if variable == "stime21":
```
all this code just to initialize a bunch of variables to -1 ??
and you should use arrays of parameters rather then stimeXX

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+
+        return report
+
+    def get_c0(self, time, zenith, range_max):
```
add a doc string explaining the parameters and units

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+            range_max_clouds = (range_max - 100.) * np.cos(zenith_radians)
+        return C_0
+
+    def apply_time_dependent_corrections(self, datime, c0, coszd, case_index=None):
```
case_index is not used in the function

add doc string with explanation of parameters

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+
+    def apply_time_dependent_corrections(self, datime, c0, coszd, case_index=None):
+        switch_times = {}
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/switch_times.txt'
```
use relative paths

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        with open(filename, 'r') as file:
+             switch_time = None
+             for line in file:
+                 if line.startswith("Switch Time:"):
```
this is a bit dangerous because it depends on the sequence of switch time and correction parameters lines.
Why not simply put both in the same line, first the date and then correction parameters
maybe also drop the lines starting with # (like you do in params.txt) and add a comment in the file

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        stimes = [(datime - switch_time).total_seconds() for switch_time in switch_times]
+        Alphafit_corr = 0.0
+
+    def apply_zenith_correction(self, c0, coszd):
```
docstring

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        return True
+
+    def apply_azimuth_correction(self, c0, zenith, azimuth):
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/zd_az_corr.txt'
```
relative paths

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        c0 += 2 * zd_corr
+        return True
+
+    def apply_azimuth_correction(self, c0, zenith, azimuth):
```
docstring

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        with open(filename, 'r') as file:
+            rule = {}
+            for line in file:
+                if line.startswith("Rule"):
```
add skipping lines with #

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+                    if rule:
+                        rules.append(rule)
+                        rule = {}
+                elif line.startswith("Zenith Threshold:"):
```
what is the difference between zenith threshold and zenith threshold min/max?

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+                    return True
+        return False
+
+    def get_bin_width(self, ifadc, bgsamples, ncollapse):
```
docstring

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        return False
+
+    def get_bin_width(self, ifadc, bgsamples, ncollapse):
+        filename = '/home/zywuckan/ctapipe_io_magic/ctapipe_io_magic/params.txt'
```
relative paths

also this file seems to be opened in multiple functions

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        if bgsamples <= gkBGSamplesV1to4:
+            return 0.0
+        if ifadc < 128:
+            return - bgsamples * gkIntegrationWindow * self.get_bin_width(ifadc, bgsamples-1, ncollapse)
```
I hope this function will not be executed for every event, because it is doing file operations

jsitarek commented on ctapipe_io_magic/laser_params_corr_test.py:
```diff
@@ -0,0 +1,411 @@
+        return ((phecounts - self.get_bin_width(ifadc, bgsamples, ncollapse) * bg) * hwcorr
+            / self.get_bin_width(ifadc, bgsamples, ncollapse))
+
+def main():
```
this is fine for tests, but ultimately you should just execute the relevant functions from __init__.py file and store the calibrated LIDAR information in the data structure (like you do with uncalibrated)

nzywucka committed:
Merge branch 'laser_test' into lidar_corr

nzywucka committed:
laser_test_new.py: improved implementation of corrections of LIDAR parameters
azimuth_conditions.txt: parameters to perform azimuth corrections
switch_times.txt and switchtimes1.txt: parameters to perform time corrections
